### PR TITLE
feat: cypress integration test

### PR DESCRIPTION
**Pull request by otlci**
Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/example/infrastructure/cypress/support/e2e.ts
Comment (DaveMadsen) [inline workbench-core/example/infrastructure/cypress/support/e2e.ts:1]: Do we need this file?

File: workbench-core/infrastructure/src/workbenchCognito.ts [hunk 1]
```diff
@@ -59,7 +59,6 @@ const userPoolClientDefaults: UserPoolClientOptions = {
   },
   authFlows: {
     userSrp: true,
-    userPassword: true,
```
Comment (DaveMadsen) [inline workbench-core/infrastructure/src/workbenchCognito.ts:62]: Have you verified this change doesn't affect SWB?

File: workbench-core/example/infrastructure/cypress/support/e2e.ts
Comment (otlci) [reply workbench-core/example/infrastructure/cypress/support/e2e.ts:1]: Yes, cypress requires this file to determine type of tests

File: workbench-core/infrastructure/src/workbenchCognito.ts (hunk 1 above)
Comment (otlci) [reply workbench-core/infrastructure/src/workbenchCognito.ts:62]: I've recreated fresh environment for SWB and integration tests are passing. Are there any additional checks that I should execute?

File: workbench-core/example/infrastructure/cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  });
+
+  return response.Parameter!.Value;
+};
+
+const getEnvironmentVariables = async (stage: string) => {
+  console.info('Stage:', stage);
+
+  const config = yaml.load(
+    fs.readFileSync(join(__dirname, `./integration-tests/config/${stage}.yaml`), 'utf8')
```
Comment (FernandoAranda) [inline workbench-core/example/infrastructure/cypress.config.ts:28]: Do we need this? 
We are basically requiring integ-test configuration in order to run e2e tests when IMO they should be independent of each other

Comment (otlci) [reply workbench-core/example/infrastructure/cypress.config.ts:28]: This is needed to retrieve the username and password of Cognito user in order to log in. The tests here are meant to be a complement to integration test suite that requires logging in with WebUI, not e2e UI tests per se. These tests are also ran from same Github action [deploy-integration-example-app.yml](https://github.com/aws-solutions/solution-spark-on-aws/pull/944/files#diff-8d34599e34eafa22976e842cd2c1438d89f2f7cce069e3f775f76eee3a028af8)

Comment (FernandoAranda) [reply workbench-core/example/infrastructure/cypress.config.ts:28]: got it, if they are mean to be coupled then its fine for me

**Review by FernandoAranda — APPROVED** [review]

**Review by insignias — APPROVED** [review]

**Pull request merged**